live_pipeline/transform/transform_functions.py

```diff
@@ -0,0 +1,170 @@
+"""libraries required for transformation functions"""
+from datetime import datetime as dt
+from os import environ
+from boto3 import client
+from dotenv import load_dotenv
+import json
+
+
+LOWER_TEMP_LIMIT = 9
+UPPER_TEMP_LIMIT = 40
+LOWER_SOIL_LIMIT = 21
+UPPER_SOIL_LIMIT = 40
+
+
+def create_dictionary_for_plant(raw_data: dict) -> dict:
```

Comment:
This function is very large in size. You may want to consider breaking it down into smaller functions. This would also help with ensuring all rows are tested against. E.g. rows 25, 34, 37, 41-44, 47, 49-52, 59-67 and 70 are not currently tested with unit tests.